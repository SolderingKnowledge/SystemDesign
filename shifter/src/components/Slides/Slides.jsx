import Slide  from "../Slide/Slide";
import forward from "../../assets/forward.svg";
import backward from "../../assets/backward.svg";
import IconButton from '@material-ui/core/IconButton';

export default function Slides (props) {

    const { sublist, onClickPrev, onClickNext, left, marginLeft, width, viewportSlideCount, screenWidth } = props;

    const slides = sublist.map((slide, idx)=> {
        return <Slide slide={slide} key={idx} />
    })

    //default: 3 slides; regular size desktops
    let viewPortWidth = width * viewportSlideCount - 20;// left and right margins: 10px + 10px;

    if (screenWidth <= 760 ){// mobile: 1 slide
        viewPortWidth = 320;
    } else if(screenWidth > 760 && screenWidth < 1150 ){// tablet: 2 slides
        viewPortWidth = 660;
    } else if(screenWidth > 1460 ) { // large desktops: 4 slides
        viewPortWidth = 1340;
    }


    return (
        <section tabIndex="0" className="slides" aria-label="shifter section">
            <IconButton onClick={onClickPrev} aria-label="slide previus" id="prev-button">
                <img src={backward} alt="Arrow left" />
            </IconButton>
            <div className="view-port" style={{width: `${viewPortWidth}px`}}>
                <ul className="list" style={{left: `${left}px`,marginLeft: `${marginLeft}px`}}>
                    {slides}
                </ul>
            </div>
            <IconButton onClick={onClickNext} aria-label="slide next" id="next-button">
                <img src={forward} alt="Arrow right" />
            </IconButton>
        </section>
    );
}
