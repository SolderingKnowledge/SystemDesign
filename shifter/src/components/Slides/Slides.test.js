import { shallow } from "enzyme";
import React from "react";
import Slides from "./Slides";


describe("should render Slides component", () => {

    const mockList = [1,2,3,4,5];
    const mockOnClickPrev = jest.fn();
    const mockOnClickNext = jest.fn();
    const component = shallow(<Slides 
        sublist={mockList}
        onClickPrev={mockOnClickPrev} 
        onClickNext={mockOnClickNext}
        left={0}
        marginLeft={0}
        width={0}
        viewportSlideCount={0}
        screenWidth={0}
    />);

    it('should match snapshot', () => {
        expect(component).toMatchSnapshot();
    })

    it("next, prev buttons should have onClick callback functions", () => {

        const onClickPrev = component.find('[id="prev-button"]');
        const onClickNext = component.find('[id="next-button"]');

        onClickPrev.simulate("click");
        onClickNext.simulate("click");

        expect(mockOnClickPrev.mock.calls.length).toEqual(1);
        expect(mockOnClickNext.mock.calls.length).toEqual(1);
    })

});