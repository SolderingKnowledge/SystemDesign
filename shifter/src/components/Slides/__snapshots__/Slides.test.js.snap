// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Slides component should match snapshot 1`] = `
<section
  aria-label="shifter section"
  className="slides"
  tabIndex="0"
>
  <WithStyles(ForwardRef(IconButton))
    aria-label="slide previus"
    id="prev-button"
    onClick={[MockFunction]}
  >
    <img
      alt="Arrow left"
      src="backward.svg"
    />
  </WithStyles(ForwardRef(IconButton))>
  <div
    className="view-port"
    style={
      Object {
        "width": "320px",
      }
    }
  >
    <ul
      className="list"
      style={
        Object {
          "left": "0px",
          "marginLeft": "0px",
        }
      }
    >
      <Slide
        key="0"
        slide={1}
      />
      <Slide
        key="1"
        slide={2}
      />
      <Slide
        key="2"
        slide={3}
      />
      <Slide
        key="3"
        slide={4}
      />
      <Slide
        key="4"
        slide={5}
      />
    </ul>
  </div>
  <WithStyles(ForwardRef(IconButton))
    aria-label="slide next"
    id="next-button"
    onClick={[MockFunction]}
  >
    <img
      alt="Arrow right"
      src="forward.svg"
    />
  </WithStyles(ForwardRef(IconButton))>
</section>
`;
